File: java/oop/CityCodeApp/src/com/techlabs/citycode/CityMap.java
package com.techlabs.citycode;

import java.util.HashMap;
import java.util.Map;

public class CityMap {
	HashMap<CityCode, CityCode> cityMap = new HashMap<CityCode, CityCode>();
	
	public void addCityCode(CityCode c){
		cityMap.put(c, c);
	}
	
	public void checkCityCode(String input){
		boolean anyMatchFound=false;
		for (Map.Entry<CityCode, CityCode> m : cityMap.entrySet()) {
			CityCode singleCityCode = (CityCode) m.getValue();
			
			if(input.equalsIgnoreCase(singleCityCode.getCityCode())){
				anyMatchFound=true;
				System.out.println(input+" stands for "+singleCityCode.getDescription());
				break;
			}else{
				anyMatchFound=false;
			}
			
		}
		if(!anyMatchFound){
			System.out.println("No Match Found.........");
		}
	}

	public void checkCityCodeWithChar(String inputChar) {
		for(Map.Entry<CityCode, CityCode> m :cityMap.entrySet()){
			CityCode singleCityCode = (CityCode) m.getValue();
			String description = singleCityCode.getDescription();
			for(int i=0;i<description.length();i++){
				char c = description.charAt(i);
				String decripchar = String.valueOf(c);
				if(inputChar.equalsIgnoreCase(decripchar)){
					System.out.println("May be you are looking for .."+singleCityCode.getDescription());
					break;
				}
			}
		}
		
	}
	
	

}

File: dotNet/csharp/CityCode/CityCode/CityMap.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace CityCode
{
    class CityMap
    {
        Dictionary<CityCode, CityCode> citymap = new Dictionary<CityCode, CityCode>();
        public void AddCityCode(CityCode c)
        {
            citymap.Add(c,c);

        }

        public void CheckCityCode(String input)
        {
            bool anyMatchFound = false;
            foreach (KeyValuePair<CityCode,CityCode> city in citymap)
            {
                if (input.Equals(city.Value.Code,StringComparison.CurrentCultureIgnoreCase))
                {
                    anyMatchFound = true;
                    Console.WriteLine(input + " stands for " + city.Value.Desciption);
                    break;
                }else
                {
                    anyMatchFound = false;
                }
            }

            if (!anyMatchFound)
            {
                Console.WriteLine("No match found......");
            }
        }

        public void CheckCityCodeWithChar(String inputChar)
        {
            foreach (KeyValuePair<CityCode,CityCode> city in citymap)
            {
                string description = city.Value.Desciption;
                for (int i=0;i<description.Length;i++)
                {
                    char c = description[i];
                    string descripchar = Convert.ToString(c);
                    if (inputChar.Equals(descripchar,StringComparison.CurrentCultureIgnoreCase))
                    {
                        Console.WriteLine("May br you are looking for.."+city.Value.Desciption);
                        break;
                    }
                }
            }
        }
    }
}
